### PR TITLE
Add more details to completions doc

### PR DESCRIPTION
**Pull request by gabritto**

### CODE REVIEW

File: codebase/src/services/completions.md [hunk 1]
```diff
@@ -1,9 +1,62 @@
...
+A completion snippet for declaring a method `foo`, with a tab stop (`${0}`) in its body:
+```ts
+{
+    isSnippet: true,
+    insertText: "foo() { ${0} }",
+}
+```
+becomes this in VSCode, when accepted (note the cursor position):
+![Screenshot of vscode with code `class Foo { foo() { | } }` in it.](../../screenshots/threeslash-refs.png)
+For more on snippets, see [Snippets in Visual Studio Code](https://code.visualstudio.com/docs/editor/userdefinedsnippets).
```
Comment (andrewbranch) [inline codebase/src/services/completions.md:37]: Preserving the list hierarchy with a little Markdown magic:

````suggestion
* **isSnippet**: if this is true, then this completion entry is a snippet, and `insertText` is a snippet text.  
  e.g.:
  A completion snippet for declaring a method `foo`, with a tab stop (`${0}`) in its body:
  ```ts
  {
      isSnippet: true,
      insertText: "foo() { ${0} }",
  }
  ```
  becomes this in VSCode, when accepted (note the cursor position):
  ![Screenshot of vscode with code `class Foo { foo() { | } }` in it.](../../screenshots/threeslash-refs.png)
  For more on snippets, see [Snippets in Visual Studio Code](https://code.visualstudio.com/docs/editor/userdefinedsnippets).
````

File: codebase/src/services/completions.md [hunk 2]
```diff
@@ -1,9 +1,62 @@
-## Completions in TS
+# Completions
 
-Completions come in through the language service, and most of the work starts in
-[`getCompletionEntriesFromSymbols`](https://github.com/Microsoft/TypeScript/blob/340f81035ff1d753e6a1f0fedc2323d169c86cc6/src/services/completions.ts#L305).
 
-### Completion Data
+Completions for TypeScript and JavaScript are provided by TypeScript's language service.
+When you are in the middle of typing something in an editor (or if you hit `Ctrl` + `Space` in VSCode), the editor sends a request to the TypeScript language service.
```
Comment (DanielRosenwasser) [inline codebase/src/services/completions.md:5]:
```suggestion
When you are in the middle of typing something in an editor (or if you hit <kbd>Ctrl</kbd> + <kbd>Space</kbd> in VSCode), the editor sends a request to the TypeScript language service.
```

File: codebase/src/services/completions.md [hunk 3]
```diff
@@ -1,9 +1,62 @@
...
+e.g.:
+A completion snippet for declaring a method `foo`, with a tab stop (`${0}`) in its body:
+```ts
+{
+    isSnippet: true,
+    insertText: "foo() { ${0} }",
+}
+```
+becomes this in VSCode, when accepted (note the cursor position):
+![Screenshot of vscode with code `class Foo { foo() { | } }` in it.](../../screenshots/threeslash-refs.png)
```
Comment (andrewbranch) [inline codebase/src/services/completions.md:36]:
```suggestion
![Screenshot of vscode with code `class Foo { foo() { | } }` in it.](../../screenshots/snippet-vscode.png)
```

**Review by andrewbranch — COMMENTED** [review]
This is a great addition 💯

**Comment by orta** [comment]
👍🏻

**Pull request merged**